### PR TITLE
Updating Operations API to support Python's AutoRest

### PR DESCRIPTION
**Pull request by sonathan**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by veronicagg — CHANGES_REQUESTED** [review]
Please review model validation errors in added example at https://travis-ci.org/Azure/azure-rest-api-specs/jobs/255998310#L661.
I understand you're not making changes in the rest of the spec, so just making sure you know about linter error/warnings being reported: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/255998308#L674. 
Thanks!

File: specification/recoveryservices/resource-manager/readme.md
Comment (sonathan) [inline specification/recoveryservices/resource-manager/readme.md]: Yes, we have requested approval for this supprestion already and also for the XmsExamplesProvidedValidation, as said by @ravbhatnagar that "As Samer said, we cant give an exception for the examples violation. This will not block the PR to be merged but it will be shown as a violation till it is fixed." we had a mail thread with azure-spc-violations@microsoft.com, adxsr@microsoft.com with subject: Violations Suppression requested for PR 1139

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json
Comment (sonathan) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json]: Thankyou. Removing SubsId and ResourceGroupName from parameter from subsequent commit.

File: specification/recoveryservices/resource-manager/readme.md
Comment (veronicagg) [inline specification/recoveryservices/resource-manager/readme.md]: @sonathan thanks for pointing this out. Regarding the specific violation DefinitionsPropertiesNamesCamelCase the justification in the email and the above doesn't seem to match the reason why the error is being reported. I see properties named as as "Provider" for "ClientDiscoveryDisplay", which violates the rule. I understand this is a breaking change and you may not want to take it right now, would this be a permanent exception? @ravbhatnagar @salameer

**Review by veronicagg — CHANGES_REQUESTED** [review]

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json
Comment (veronicagg) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json]: thanks, will take a look in your next commit.

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json
Comment (sonathan) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json]: For this you can refer: https://github.com/Azure/azure-rest-api-specs/pull/1267 CLI is not yet released and it will be including these changes.

Comment (ravbhatnagar) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json]: @veronicagg - This fix is fine since otherwise the API had no meaning and clients couldnt anyway meaningfully use it. And as @sonathan pointed out, the CLI is not yet released so from that perspective it should be ok as well.

File: specification/recoveryservices/resource-manager/readme.md
Comment (ravbhatnagar) [inline specification/recoveryservices/resource-manager/readme.md]: @sonathan - We cant add suppression for the camel casing violation. Till this is fixed, it will continue to show as violation.

**Comment by ravbhatnagar** [comment]
@veronicagg - As stated above, lets not add suppression for the camel case fix. It will remain a violation as the service team is expected to fix it.

**Comment by veronicagg** [comment]
thanks @ravbhatnagar .
@sonathan please remove suppression that was added to the readme file, then I can approve this PR. thanks!

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/examples/ListOperations.json
Comment (veronicagg) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/examples/ListOperations.json]: There is a remaining issue with the example model validation https://travis-ci.org/Azure/azure-rest-api-specs/jobs/259348817#L657 
The value for blobDuration doesn't match the date-time format per https://xml2rfc.tools.ietf.org/public/rfc/html/rfc3339.html#anchor14
Please update these.

**Review by veronicagg — CHANGES_REQUESTED** [review]

**Comment by veronicagg** [comment]
@sonathan just let me know if you have questions about the issue pointed out above.

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/examples/ListOperations.json
Comment (sonathan) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/examples/ListOperations.json]: Updated. Thanks Veronica!

**Comment by sonathan** [comment]
@veronicagg I have updated the examples.

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json
Comment (veronicagg) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json]: is the field really not a date-time?

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/examples/ListOperations.json
Comment (veronicagg) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/examples/ListOperations.json]: why did you decide to remove this part? It's good to include more and cover more fields, we need to make sure the wire matches what's in the swagger, that's the purpose of having the example covering more.

**Review by veronicagg — CHANGES_REQUESTED** [review]
questions inline

**Comment by ravbhatnagar** [comment]
@veronicagg - I see the label indicating WaitForARMFeedback. Just wanted to confirm if anything is needed from my side on this.

**Comment by veronicagg** [comment]
@ravbhatnagar no, you've commented on the suppression not being allowed above, and  the label did not get removed then. I'll remove it now, next time, feel free to remove it after you provide your feedback. thanks!
On this PR I'm waiting on @sonathan regarding my questions on the examples.

**Comment by sonathan** [comment]
Hi @veronicagg 

As we are in preview, and the contracts had changed in service side, I have updated those here and have also updated examples to cover out as much scenarios as possible.

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json
Comment (veronicagg) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json]: Please review: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/262193673#L1444

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json
Comment (veronicagg) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json]: Please review: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/262193673#L1456

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json
Comment (veronicagg) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json]: Please review: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/262193673#L1469

**Review by veronicagg — CHANGES_REQUESTED** [review]

**Comment by veronicagg** [comment]
@sonathan one more question, since you mentioned you're in preview...
1) are the changes included in this spec deployed?
2) is there a reason why your api-version is not updated or includes "preview" in it?

**Comment by sonathan** [comment]
@veronicagg  Yes the changes are already deployed. To be honest we have added this API only for compliance, whenever there is a need we refer to recoveryServicesBackup for this API.

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json
Comment (sonathan) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json]: Hi @veronicagg , As this field is in Response and Shoebox creates blobs in customer storage account every hour. The value PT1H indicates blobs per hour. Thanks!

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json
Comment (sonathan) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json]: Thanks for pointing out.

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json
Comment (sonathan) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json]: Thanks!

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json
Comment (veronicagg) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json]: Thanks @sonathan . Should this be part of its description? Are customers expected to know what it means, if so, how? thanks!

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json
Comment (veronicagg) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json]: @sonathan it appears that this one is still showing up https://travis-ci.org/Azure/azure-rest-api-specs/jobs/262537716#L1457 therefore causing 1 more error in the Linter compared to the spec before this PR, you may want to update this one as well.

**Review by veronicagg — CHANGES_REQUESTED** [review]
Looks good, just one question left and one linter error that appears to not have been updated in the last commit.

**Comment by veronicagg** [comment]
@sonathan if you have any questions about my last comment, please let me know, thanks.

**Comment by veronicagg** [comment]
@sonathan any updates?

**Comment by veronicagg** [comment]
@sonathan FYI - the PR will get closed in a few days if there's no activity from your side. It it gets closed, you may reopen it when you're back to it. Thanks!

**Comment by veronicagg** [comment]
@sonathan This PR will get closed tomorrow, if no updates are received, thanks.

**Comment by veronicagg** [comment]
@sonathan please reopen the PR and tag me when you're back to this, thanks.

**Pull request closed without merging**
